pint_pandas/pint_array.py

```diff
@@ -221,13 +221,6 @@ def __init__(self, values, dtype=None, copy=False):
         if not isinstance(values, np.ndarray):
             values = np.array(values, copy=copy)
             copy = False
-        if not np.issubdtype(values.dtype, np.floating):
-            warnings.warn(
-                f"pint-pandas does not support magnitudes of {values.dtype}. Converting magnitudes to float.",
-                category=RuntimeWarning,
-            )
```

Comment:
Don't delete this. Integers still need to be converted to float or need another workaround.

Reply:
Why would you want to convert integers to float? Floats can not represent the same range. What is the problem that needs working around?
Nonscalar values (objects, tuples, complex numbers) will all trigger this, which make PintArray not able to represent arrays of all possible pint Quantity:s.

Reply:
Once upon a time, only floats could pun NA data as np.nan, and This Was The Way for handling NA data.  But later Pandas developed nullable ints, at which point that should also have become an option.  Nowadays, the EA implementation has first-class pd.NA handling, and expect that when we finally put weight on that, it will hold.  But I don't have the years of experience that others do, so may well be missing some important details.

Reply:
I changed the code to back the PintArray by a pd.array, rather than a np.ndarray, and map numpy integer dtypes to Pandas ones, so that <NA> is correctly handled also by PintArray. Hope this is a good enough solution :)

Reply:
This is a great solution!

Reply:
:)